backend/codesirius/internal_api/views/apikey.py

```
@@ -0,0 +1,53 @@
+
+    def delete(self, request: Request, api_key_id: int) -> CodesiriusAPIResponse:
+        try:
+            api_key = APIKey.objects.get(id=api_key_id)
+        except APIKey.DoesNotExist:
+            raise NotFound("API key not found")
+        api_key.delete()
+        return CodesiriusAPIResponse(message="API key deleted")
+
+
+class ProtectedAPIView(APIView):
+    authentication_classes = [APIAuthentication]
```

Ruff (0.8.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)